allow NAs again (for real now)

dpprdan opened the pull request:
This PR adds support for `NA`s in placenames and latitude longitude again. Queries with `NA`s are not sent to the API. An empty/`NA` response will be returned depending on the specific query.

dpprdan committed:
drop !dplyr::between()

dpprdan committed:
allow NAs for placename or latitude/longitude again

dpprdan committed:
set `roadinfo = NULL` in `.oc_process()`

dpprdan committed:
test `oc_check_query()` with latitude or longitude out of bounds

dpprdan committed:
simplify package docs

roxygen fetches info from DESCRIPTION

dpprdan committed:
add NEWS item re `NA`s

dpprdan committed:
make lintr happy

dpprdan commented:
The codecov warnings are false positives due to the missing key on PRs (probably should disable those tests for PRs for now). I get 98.30% coverage locally.

dpprdan committed:
disable coverage workflow for PRs

Many tests will not run on PRs, because the opencage key on GitHub Actions is not available for PRs.
Therefore the reported coverage is not valid on PRs.

maelle commented on R/oc_bbox.R:
```diff
@@ -131,16 +131,16 @@ oc_check_bbox <- function(xmin, ymin, xmax, ymax) {
   if (!all(is.numeric(bnds))) {
     stop("Every `bbox` must be a numeric vector.", call. = FALSE)
   }
-  if (!dplyr::between(xmin, -180, 180)) {
```
why not an helper function for that? `opencage_between()` or so?

dpprdan replied:
I made this its own check function... 🤷🏻‍♂️

maelle reviewed (approved):
LGTM! Is it an use case you encounter? (I guess so :sweat_smile: )

dpprdan committed:
add `oc_check_between()`

dpprdan committed:
use roxygen 7.2.0

dpprdan commented:
> Is it an use case you encounter?

Sometimes I might have a not-so-clean dataset that I want to geocode quickly. Then it helps if it doesn't abort the whole thing in case there are only a few NAs...

dpprdan committed:
test `oc_check_between()` error messages and with non-numeric

maelle commented on tests/testthat/test-oc_check_query.R:
```diff
@@ -257,3 +257,17 @@ test_that("oc_check_logical checks no_record", {
+test_that("oc_check_logical checks no_record", {
+  # numeric()
+  symbol <- 11
+  expect_error(oc_check_between(symbol, 0, 10))
```
you could use `expect_snapshot_error()`.

dpprdan replied:
using `expect_snapshot(error = TRUE)` now, so the snap file also contains the code and not only the error message (which it does when using `expect_snapshot_error()`

maelle reviewed (approved):
:rocket:

dpprdan committed:
use `expect_snapshot()` in `oc_check_between()` tests

dpprdan committed:
update README

- switch to badges.cranchecks.info
- polish badges Alt text
- use kable for printing tables

dpprdan committed:
polish NEWS
